crates/bevy_winit/src/state.rs

```diff
@@ -0,0 +1,762 @@
+use approx::relative_eq;
+use bevy_app::{App, AppExit, PluginsState};
+use bevy_ecs::change_detection::{DetectChanges, NonSendMut, Res};
+use bevy_ecs::entity::Entity;
+use bevy_ecs::event::{EventWriter, ManualEventReader};
+use bevy_ecs::prelude::*;
+use bevy_ecs::system::SystemState;
+use bevy_ecs::world::FromWorld;
+use bevy_input::{
+    mouse::{MouseButtonInput, MouseMotion, MouseScrollUnit, MouseWheel},
+    touchpad::{TouchpadMagnify, TouchpadRotate},
+};
+use bevy_log::{error, trace, warn};
+use bevy_math::{ivec2, DVec2, Vec2};
+#[cfg(not(target_arch = "wasm32"))]
+use bevy_tasks::tick_global_task_pools_on_main_thread;
+use bevy_utils::Instant;
+use std::marker::PhantomData;
+use winit::application::ApplicationHandler;
+use winit::dpi::{LogicalSize, PhysicalSize};
+use winit::event;
+use winit::event::{DeviceEvent, DeviceId, StartCause, WindowEvent};
+use winit::event_loop::{ActiveEventLoop, ControlFlow, EventLoop};
+use winit::window::WindowId;
+
+#[allow(deprecated)]
+use bevy_window::{
+    AppLifecycle, CursorEntered, CursorLeft, CursorMoved, FileDragAndDrop, Ime, ReceivedCharacter,
+    RequestRedraw, Window, WindowBackendScaleFactorChanged, WindowCloseRequested, WindowDestroyed,
+    WindowFocused, WindowMoved, WindowOccluded, WindowResized, WindowScaleFactorChanged,
+    WindowThemeChanged,
+};
+#[cfg(target_os = "android")]
+use bevy_window::{PrimaryWindow, RawHandleWrapper};
+
+use crate::accessibility::AccessKitAdapters;
+use crate::system::CachedWindow;
+use crate::{
+    converters, create_windows, react_to_resize, AppSendEvent, CreateWindowParams, UpdateMode,
+    WinitEvent, WinitSettings, WinitWindows,
+};
+
+/// Persistent state that is used to run the [`App`] according to the current
```

Comment:
I'd like information about where this data is stored and how users can access it.

Reply:
I removed the `pub` from the struct, making it not really usable from users. Do you have any suggestions or similar comments that I can be inspired from? This is just an implementation detail now.